src/repositories/movie_repository.go

```diff
@@ -1,42 +1,296 @@
+			table.Movies.Rating.SET(mysql.Float(*movie.Rating)),
+			table.Movies.ReleaseDate.SET(mysql.DateT(movie.ReleaseDate)),
+			table.Movies.TimeInMin.SET(mysql.Int32(movie.TimeInMin)),
+			table.Movies.Fsk.SET(mysql.Int32(movie.Fsk)),
+		).WHERE(
+		table.Movies.ID.EQ(utils.MysqlUuid(movie.ID)),
+	)
+
+	// Execute the query
+	rows, err := updateQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
+}
+
+func (mr *MovieRepository) DeleteMovie(movieId *uuid.UUID) *models.KTSError {
+	// Create the delete statement
+	deleteQuery := table.Movies.DELETE().WHERE(table.Movies.ID.EQ(utils.MysqlUuid(movieId)))
+
+	// Execute the query
+	rows, err := deleteQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
+}
+
+// One Movie with all Genres
+func (mr *MovieRepository) GetMovieByIdWithGenre(movieId *uuid.UUID) (*models.MovieWithGenres, *models.KTSError) {
+	var movie models.MovieWithGenres
+
+	// Create the query
+	stmt := mysql.SELECT(
+		table.Movies.AllColumns,
+		table.Genres.AllColumns,
+	).FROM(
+		table.MovieGenres.
+			INNER_JOIN(table.Movies, table.Movies.ID.EQ(table.MovieGenres.MovieID)).
+			INNER_JOIN(table.Genres, table.Genres.ID.EQ(table.MovieGenres.GenreID)),
+	).WHERE(
+		table.Movies.ID.EQ(utils.MysqlUuid(movieId)),
+	)
+
+	// Execute the query
+	err := stmt.Query(mr.DatabaseManager.GetDatabaseConnection(), &movie)
+	if err != nil {
+		if err.Error() == "qrm: no rows in result set" {
+			return nil, kts_errors.KTS_MOVIE_NOT_FOUND
+		}
+		return nil, kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	return &movie, nil
+}
+
+// All Movies with all Genres - Grouped by Movie
+func (mr *MovieRepository) GetMoviesWithGenres() (*[]models.MovieWithGenres, *models.KTSError) {
+	var moviesWithGenres []models.MovieWithGenres
+
+	// Create the query
+	stmt := mysql.SELECT(
+		table.Movies.AllColumns,
+		table.Genres.AllColumns,
+	).FROM(
+		table.MovieGenres.
+			INNER_JOIN(table.Movies, table.Movies.ID.EQ(table.MovieGenres.MovieID)).
+			INNER_JOIN(table.Genres, table.Genres.ID.EQ(table.MovieGenres.GenreID)),
+	)
+
+	// Execute the query
+	err := stmt.Query(mr.DatabaseManager.GetDatabaseConnection(), &moviesWithGenres)
+	if err != nil {
+		return nil, kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if len(moviesWithGenres) == 0 {
+		return nil, kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return &moviesWithGenres, nil
+}
+
+func (mr *MovieRepository) GetMovieByIdWithEverything(movieId *uuid.UUID) (*models.MovieWithEverything, *models.KTSError) {
```

Comment:
Yea this and then remove the others ig

Reply:
The others might be needed for fast search / filter

Reply:
Because not always everything is needed

Reply:
When would they query a single movie and not need everything?

Reply:
Fair enough. I get it now

Reply:
Updated to GetMovieById